include/itkAttenuationImageFilter.hxx

```diff
@@ -172,26 +209,38 @@ AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::ThreadedGenerateD
         end[m_Direction] -= m_PadUpperBounds;
 
         // Estimate attenuation for each inclusion pixel
-        // with respect to the last pixel in the inclusion
-        while (start[m_Direction] < end[m_Direction])
-        {
-          // If no fixed estimation depth is set or too few pixels remain for fixed-depth estimation
-          // then take the attenuation between the given pixel and the last pixel in the inclusion
-          InputIndexType target = end;
+        // by weighted average of pair-wise attenuations for all pairs
+        unsigned accumulationStartIndex = regionForThread.GetIndex(m_Direction);
+        std::fill_n(
+          &accumulatedWeight[start[m_Direction] - accumulationStartIndex], end[m_Direction] - start[m_Direction], 0.0f);
 
-          if (m_FixedEstimationDepth != 0 && end[m_Direction] - start[m_Direction] > m_FixedEstimationDepth)
+        while (start[m_Direction] <= end[m_Direction])
+        {
+          for (IndexValueType k = start[m_Direction] + 1; k <= end[m_Direction]; ++k)
           {
-            target[m_Direction] = start[m_Direction] + m_FixedEstimationDepth;
+            unsigned pixelDistance = k - start[m_Direction];
+
+            InputIndexType target = start;
+            target[m_Direction] = k;
+            float estimatedAttenuation = ComputeAttenuation(target, start);
+            float weight = pixelDistance < distanceWeightsSize ? m_DistanceWeights[pixelDistance] : 1.0f;
```

Comment:
Could you please add explanation on this line? I'm not sure I understand how the `1.0f` weight fits into the rest of the algorithm.

Reply:
The pairing pixels nearby have low weight. The further pixels have high weight, which approaches 1.0. If the distance is greater than the size of our distance weights, it should be the limit (`1.0f`).